src/server/main_service.cc

```diff
@@ -2097,19 +2122,12 @@ void Service::Exec(CmdArgList args, ConnectionContext* cntx) {
     }
   }
 
-  if (exec_info.preborrowed_interpreter) {
-    // Use SafeTLocal() to avoid accessing the wrong thread local instance
-    ServerState::SafeTLocal()->ReturnInterpreter(exec_info.preborrowed_interpreter);
```

Comment:
I don't use safe tlocal, because if we migrated, we're already returning it to the wrong pool... We need to guard against this somehow, currently we don't migrate in commands (except replica stuff), so high level guards are optimal

Reply:
Added a pointer dcheck